aardvark-app/src/application.rs

```diff
@@ -150,8 +149,15 @@ impl AardvarkApplication {
                     .expect("rx should be given at this point");
 
                 glib::spawn_future_local(async move {
-                    while let Some(bytes) = rx.recv().await {
-                        application.ingest_message(bytes);
+                    while let Some(message) = rx.recv().await {
+                        match message {
+                            ToApp::NewDocument(text_document) => {
+                                // @TODO: get the short code for this document into the share UI
```

Comment:
Need to send the short code to the backend here.

Reply:
Another approach (maybe nicer) for this messaging channel would be to still only have one message variant, but always include the document id as well as the message bytes. Then when a change in document id is detected we interpret this as a change of document.

Doesn't change anything about how the frontend would react, just how this change is communicated to the front end.